e2e2/internal/framework_extensions/client.go

```diff
@@ -114,6 +122,62 @@ func RenderManifests(file []byte, templateData interface{}) ([]byte, error) {
 	return buf.Bytes(), err
 }
 
+// GetMPIJobLogs get logs from MPIJob
+func GetMPIJobLogs(ctx context.Context, restConfig *rest.Config, job *kubeflowv2beta1.MPIJob) (string, error) {
+	clientset, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return "", err
+	}
+	pods, err := clientset.CoreV1().Pods(job.Namespace).List(ctx, metav1.ListOptions{LabelSelector: fmt.Sprintf("job-name=%s-launcher", job.Name)})
+	if err != nil {
+		return "", err
+	}
+	if len(pods.Items) == 0 {
+		return "", fmt.Errorf("no pods found for job %s", job.Name)
+	}
+	log := clientset.CoreV1().Pods(job.Namespace).GetLogs(pods.Items[0].Name, &corev1.PodLogOptions{})
+	podLogs, err := log.Stream(ctx)
+	if err != nil {
+		return "", err
+	}
+	defer podLogs.Close()
+	buf := new(bytes.Buffer)
+	_, err = io.Copy(buf, podLogs)
+	if err != nil {
+		return "", err
+	}
+	str := buf.String()
+	return str, nil
+}
+
+// GetMPIJobLogs get logs from MPIJob
+func GetJobLogs(ctx context.Context, restConfig *rest.Config, job *batchv1.Job) (string, error) {
```

Comment:
the 2 method looks similar to each other, can we reduce the duplication somehow?

Reply:
The job name is different `job-name=xxx-launcher` vs `job-name=xxx`

Reply:
can the job-name be another parameter or get from the job itself?

Reply:
Maybe something like this?
```
func GetJobLogs(ctx context.Context, restConfig *rest.Config, job interface{}) error {
    switch job := job.(type) {
    case *batchv1.Job:
      // assign job name
      // ...
    case *kubeflowv2beta1.MPIJob:
        // assign job name
        // ...
    default:
        return fmt.Errorf("unsupported job type: %T", job)
// proceed with fetching log
    }
}
``